### ISSUE TITLE
Duplicate GlobalKey detected in widget tree.

### ISSUE THREAD

**Issue by kalptech2021** [GetStream/stream-chat-flutter#1576]
### Which packages are you using?

stream_chat_flutter

### On what platforms did you experience the issue?

iOS, Android

### What version are you using?

  stream_chat_flutter: ^6.1.0

### What happened?

```terminal
======== Exception caught by widgets library =======================================================
The following assertion was thrown building RawGestureDetector-[LabeledGlobalKey<RawGestureDetectorState>#c25d8](state: RawGestureDetectorState#7db3a(gestures: [vertical drag], behavior: opaque)):
'package:flutter/src/widgets/framework.dart': Failed assertion: line 6427 pos 12: 'child == _child': is not true.


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md

The relevant error-causing widget was: 
  UnboundedCustomScrollView UnboundedCustomScrollView:file:///Users/amandoraya/.pub-cache/hosted/pub.dev/stream_chat_flutter-6.1.0/lib/scrollable_positioned_list/src/positioned_list.dart:189:16
When the exception was thrown, this was the stack: 
#2      SingleChildRenderObjectElement.forgetChild (package:flutter/src/widgets/framework.dart:6427:12)
#3      Element._retakeInactiveElement (package:flutter/src/widgets/framework.dart:3907:14)
...     Normal element mounting (10 frames)
#13     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3971:16)
#14     Element.updateChild (package:flutter/src/widgets/framework.dart:3702:20)
#15     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#16     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5251:11)
#17     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#18     StatefulElement.update (package:flutter/src/widgets/framework.dart:5274:5)
#19     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#20     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6442:14)
#21     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#22     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#23     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#24     ProxyElement.update (package:flutter/src/widgets/framework.dart:5417:5)
#25     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#26     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6442:14)
#27     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#28     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#29     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#30     ProxyElement.update (package:flutter/src/widgets/framework.dart:5417:5)
#31     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#32     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6442:14)
#33     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#34     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6442:14)
#35     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#36     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6442:14)
#37     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#38     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#39     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#40     ProxyElement.update (package:flutter/src/widgets/framework.dart:5417:5)
#41     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#42     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#43     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5251:11)
#44     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#45     StatefulElement.update (package:flutter/src/widgets/framework.dart:5274:5)
#46     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#47     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#48     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5251:11)
#49     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#50     StatefulElement.update (package:flutter/src/widgets/framework.dart:5274:5)
#51     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#52     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#53     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#54     ProxyElement.update (package:flutter/src/widgets/framework.dart:5417:5)
#55     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#56     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#57     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#58     StatelessElement.update (package:flutter/src/widgets/framework.dart:5162:5)
#59     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#60     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#61     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#62     ProxyElement.update (package:flutter/src/widgets/framework.dart:5417:5)
#63     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#64     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#65     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5251:11)
#66     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#67     StatefulElement.update (package:flutter/src/widgets/framework.dart:5274:5)
#68     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#69     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#70     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5251:11)
#71     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#72     StatefulElement.update (package:flutter/src/widgets/framework.dart:5274:5)
#73     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#74     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#75     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#76     ProxyElement.update (package:flutter/src/widgets/framework.dart:5417:5)
#77     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#78     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6442:14)
#79     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#80     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#81     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#82     StatelessElement.update (package:flutter/src/widgets/framework.dart:5162:5)
#83     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#84     RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:6093:32)
#85     MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6595:17)
#86     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#87     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6442:14)
#88     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#89     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#90     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5251:11)
#91     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#92     StatefulElement.update (package:flutter/src/widgets/framework.dart:5274:5)
#93     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#94     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5111:16)
#95     Element.rebuild (package:flutter/src/widgets/framework.dart:4805:7)
#96     StatelessElement.update (package:flutter/src/widgets/framework.dart:5162:5)
#97     Element.updateChild (package:flutter/src/widgets/framework.dart:3686:15)
#98     _LayoutBuilderElement._layout.layoutCallback (package:flutter/src/widgets/layout_builder.dart:135:18)
#99     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2720:19)
#100    _LayoutBuilderElement._layout (package:flutter/src/widgets/layout_builder.dart:153:12)
#101    RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:2506:59)
#102    PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:1062:15)
#103    RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:2506:14)
#104    RenderConstrainedLayoutBuilder.rebuildIfNecessary (package:flutter/src/widgets/layout_builder.dart:228:7)
#105    _RenderLayoutBuilder.performLayout (package:flutter/src/widgets/layout_builder.dart:313:5)
#106    RenderObject.layout (package:flutter/src/rendering/object.dart:2395:7)
#107    RenderBox.layout (package:flutter/src/rendering/box.dart:2386:11)
#108    RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:122:14)
#109    EnhancedRenderLeaderLayer.performLayout (package:flutter_portal/src/enhanced_composited_transform/flutter_src/rendering_proxy_box.dart:73:11)
#110    RenderObject.layout (package:flutter/src/rendering/object.dart:2395:7)
#111    RenderBox.layout (package:flutter/src/rendering/box.dart:2386:11)
#112    ChildLayoutHelper.layoutChild (package:flutter/src/rendering/layout_helper.dart:52:11)
#113    RenderStack._computeSize (package:flutter/src/rendering/stack.dart:580:43)
#114    RenderStack.performLayout (package:flutter/src/rendering/stack.dart:607:12)
#115    RenderObject.layout (package:flutter/src/rendering/object.dart:2395:7)
#116    RenderBox.layout (package:flutter/src/rendering/box.dart:2386:11)
#117    ChildLayoutHelper.layoutChild (package:flutter/src/rendering/layout_helper.dart:52:11)
#118    RenderStack._computeSize (package:flutter/src/rendering/stack.dart:580:43)
#119    RenderStack.performLayout (package:flutter/src/rendering/stack.dart:607:12)
#120    RenderObject.layout (package:flutter/src/rendering/object.dart:2395:7)
#121    RenderBox.layout (package:flutter/src/rendering/box.dart:2386:11)
#122    RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:122:14)
#123    RenderObject.layout (package:flutter/src/rendering/object.dart:2395:7)
#124    RenderBox.layout (package:flutter/src/rendering/box.dart:2386:11)
#125    RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:122:14)
#126    RenderObject.layout (package:flutter/src/rendering/object.dart:2395:7)
#127    RenderBox.layout (package:flutter/src/rendering/box.dart:2386:11)
#128    ChildLayoutHelper.layoutChild (package:flutter/src/rendering/layout_helper.dart:52:11)
#129    RenderFlex._computeSizes (package:flutter/src/rendering/flex.dart:868:45)
#130    RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:903:32)
#131    RenderObject.layout (package:flutter/src/rendering/object.dart:2395:7)
#132    RenderBox.layout (package:flutter/src/rendering/box.dart:2386:11)
#133    MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:171:12)
#134    _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:1056:7)
#135    MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:235:7)
#136    RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:403:14)
#137    RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:2234:7)
#138    PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:1016:18)
#139    RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:492:19)
#140    WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:905:13)
#141    RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:358:5)
#142    SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1284:15)
#143    SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1214:9)
#144    SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1072:5)
#145    _invoke (dart:ui/hooks.dart:142:13)
#146    PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:359:5)
#147    _drawFrame (dart:ui/hooks.dart:112:31)
(elided 2 frames from class _AssertionError)
====================================================================================================

======== Exception caught by widgets library =======================================================
The following assertion was thrown while finalizing the widget tree:
Duplicate GlobalKey detected in widget tree.

The following GlobalKey was specified multiple times in the widget tree. This will lead to parts of the widget tree being truncated unexpectedly, because the second time a key is seen, the previous instance is moved to the new location. The key was:
- [GlobalKey#5acec]
This was determined by noticing that after the widget with the above global key was moved out of its previous parent, that previous parent never updated during this frame, meaning that it either did not update at all or updated before the widget was moved, in either case implying that it still thinks that it should have a child with that global key.
The specific parent that did not update after having one or more children forcibly removed due to GlobalKey reparenting is:
- Semantics(container: false, properties: SemanticsProperties, tooltip: null, renderObject: RenderSemanticsAnnotations#ca165 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE DETACHED)
A GlobalKey can only be specified on one widget at a time in the widget tree.
When the exception was thrown, this was the stack: 
#0      BuildOwner.finalizeTree.<anonymous closure> (package:flutter/src/widgets/framework.dart:3110:15)
#1      BuildOwner.finalizeTree (package:flutter/src/widgets/framework.dart:3135:8)
#2      WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:906:19)
#3      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:358:5)
#4      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1284:15)
#5      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1214:9)
#6      SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1072:5)
#7      _invoke (dart:ui/hooks.dart:142:13)
#8      PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:359:5)
#9      _drawFrame (dart:ui/hooks.dart:112:31)
====================================================================================================
```

### Steps to reproduce

```bash
1. Send 20+ messages to a channel
2. Open that channel
3. Dont click on scroll to bottom, Type and send a message on the channel, without reading all messages upto the bottom.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kalptech2021** [comment]
Upon furthur investigation
![Screenshot 2023-05-29 at 8 29 33 PM](https://github.com/GetStream/stream-chat-flutter/assets/77516989/ee28b0aa-3d79-43f7-93b5-6e90832beeed)

Also this error only comes when there are unread messages at the bottom, and we try to send a message

**Comment by xsahil03x** [comment]
Hey @kalptech2021 , looking into this.

**Comment by xsahil03x** [comment]
Hey @kalptech2021, can you test with the linked PR?

**Comment by kalptech2021** [comment]
@xsahil03x I can confirm issue is fixed now in commit a14731f, I am using this one for now, until the package has been published with the fixes. Thank you for the quick fix.
